flink/v1.13/flink/src/main/java/org/apache/iceberg/flink/sink/PartitionedDeltaWriter.java

```diff
@@ -50,17 +55,43 @@
                          Schema schema,
                          RowType flinkSchema,
                          List<Integer> equalityFieldIds,
-                         boolean upsert) {
+                         boolean upsert,
+                         Map<String, String> properties) {
     super(spec, format, appenderFactory, fileFactory, io, targetFileSize, schema, flinkSchema, equalityFieldIds,
         upsert);
     this.partitionKey = new PartitionKey(spec, schema);
+    int writersCacheSize = PropertyUtil.propertyAsInt(
+        properties,
+        TableProperties.PARTITIONED_DELTA_WRITERS_CACHE_SIZE,
+        TableProperties.PARTITIONED_DELTA_WRITERS_CACHE_SIZE_DEFAULT);
+    long evictionTimeout = PropertyUtil.propertyAsLong(
+        properties,
+        TableProperties.PARTITIONED_DELTA_WRITERS_CACHE_EVICT_MS,
+        TableProperties.PARTITIONED_DELTA_WRITERS_CACHE_EVICT_MS_DEFAULT);
+    initWritersCache(writersCacheSize, evictionTimeout);
+  }
+
+  private synchronized void initWritersCache(int writersCacheSize, long evictionTimeout) {
```

Comment:
Why is this synchronized? Isn't this only called from the constructor?

Reply:
Yes, this is a legacy modification from before, I think it can be removed